h2d/HtmlText.hx

```diff
@@ -137,39 +250,130 @@ class HtmlText extends Text {
 				font = loadFont("italic");
 			default:
 			}
-			sizes.push(len);
 			for( child in e )
-				buildSizes(child, font, sizes, forSplit);
+				buildSizes(child, font, lines, heights, baseLines, splitNode);
 			switch( nodeName ) {
 			case "p":
-				sizes.push( -1);// break
-				newLine = true;
+				if ( !newLine ) {
+					makeLineBreak();
+				}
 			default:
 			}
 		} else {
 			newLine = false;
 			var text = htmlToText(e.nodeValue);
-			var xp = 0.;
-			for( i in 0...text.length ) {
+			var leftMargin = lines.length == 0 ? 0 : lines.pop();

```

Comment:
for example here I find the `pop` on lines and not other arrays a bit worrying, but I trust that the logic is right ^^

Reply:
It's removed from the array because it gets modified during processing and inserted back (or in case if word-wrap occurs). There is no need to pop other metrics, as line remains here, only width is changed, and if line-break happens, those metrics are populated with appropriate values.